File: backend/app/schemas/team.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class TeamMemberInfo(BaseModel):
    id: int
    full_name: Optional[str]
    email: str
    current_rank: str
    registration_date: datetime
    is_verified: bool
    is_active: bool
    activity_status: str
    personal_turnover: float
    team_size: int
    depth: int
    
    class Config:
        from_attributes = True

class TeamTreeNode(BaseModel):
    id: int
    full_name: Optional[str]
    email: str
    current_rank: str
    registration_date: datetime
    is_active: bool
    personal_turnover: float
    team_turnover: float
    direct_children_count: int
    total_descendants: int
    children: List['TeamTreeNode'] = []
    
    class Config:
        from_attributes = True

class TeamStats(BaseModel):
    total_team_size: int
    first_line_count: int
    active_members: int
    inactive_members: int
    total_team_turnover: float
    
class LegStats(BaseModel):
    member_id: int
    member_name: Optional[str]
    team_size: int
    turnover: float
    percentage: float

class TeamLegBreakdown(BaseModel):
    first_leg: LegStats
    second_leg: LegStats
    other_legs_combined: LegStats
    all_legs: List[LegStats]

File: frontend/src/types/team.ts
export interface TeamMemberInfo {
  id: number;
  full_name?: string;
  email: string;
  current_rank: string;
  registration_date: string;
  is_verified: boolean;
  is_active: boolean;
  activity_status: string;
  personal_turnover: number;
  team_size: number;
  depth: number;
}

export interface TeamTreeNode {
  id: number;
  full_name?: string;
  email: string;
  current_rank: string;
  registration_date: string;
  is_active: boolean;
  personal_turnover: number;
  team_turnover: number;
  direct_children_count: number;
  total_descendants: number;
  children: TeamTreeNode[];
}

export interface TeamStats {
  total_team_size: number;
  first_line_count: number;
  active_members: number;
  inactive_members: number;
  total_team_turnover: number;
}

export interface LegStats {
  member_id: number;
  member_name?: string;
  team_size: number;
  turnover: number;
  percentage: number;
}

export interface TeamLegBreakdown {
  first_leg: LegStats;
  second_leg: LegStats;
  other_legs_combined: LegStats;
  all_legs: LegStats[];
}
